frontend/src/pages/home/home.tsx

```diff
@@ -77,73 +81,86 @@ export default class HomePage extends React.Component<HomePageProps, HomePageSta
-            })
-            .catch((error: Error) => {
-                console.error(error);
-            });
-        const cachedArtworks: Artwork[] | null = SessionService.getGallery();
-        if (cachedArtworks && cachedArtworks.length) {
-            this.setState({artloading: false, artworks: cachedArtworks});
+            .then(setAnnouncementsToState)
+            .catch(console.error);
+    }
+
+    async loadVideo() {
+        const setVideosToState = (videos: Video[]) => this.setState({ videos });
+        const getVideoFromService = () => this.manoAloeService.getVideo()
+            .then(setVideosToState)
+            .catch(console.error);
+
+        const videos = SessionService.getVideo() ?? [];
+        if (videos?.length) {
+            setVideosToState(videos);
         } else {
-            this.manoAloeService.getGallery()
-                .then((artworks: Artwork[]) => {
-                    SessionService.saveGallery(artworks);
-                    this.setState({artloading: false, artworks});
-                })
-                .catch((error: Error) => {
-                    console.error(error);
-                })
+            getVideoFromService().finally(
+                () => SessionService.saveVideo(this.state.videos)
+            );
         }
-        const cachedVideos: Video[] | null = SessionService.getVideo();
-        if (cachedVideos && cachedVideos.length) {
-            this.setState({videos: cachedVideos});
+    }
+
+
+    async loadArtwork() {
+        const setArtworksToState = (artworks: Artwork[]) => this.setState({ artloading: false, artworks });
+        const getArtworkFromService = () => this.manoAloeService.getGallery()
+            .then(setArtworksToState)
+            .catch(console.error);
+
+        const artworks = SessionService.getGallery() ?? [];
+        if (artworks?.length) {
+            setArtworksToState(artworks);
         } else {
-            this.manoAloeService.getVideo()
-                .then((videos: Video[]) => {
-                    SessionService.saveVideo(videos);
-                    this.setState({videos});
-                })
-                .catch((error: Error) => {
-                    console.error(error);
-                })
+            getArtworkFromService().finally(
+                () => SessionService.saveGallery(this.state.artworks)
+            );
         }
     }
 
-    renderCardSection(data: (Message|Artwork|Video)[]) {
+    renderCardSection(data: (Message | Artwork | Video)[]) {
         return (
             <div>
                 <div className="wrapper-overlay">
-                    {this.state.messageLoaded && this.state.announcementLoaded ? <ComboSection data={data}/> : <div/>}
+                    {this.state.messageLoaded && this.state.announcementLoaded ? <ComboSection data={data} /> : <div />}
                 </div>
             </div>
         )
     }
 
     compileCardData() {
```

Comment:
If things are async, would we run into any race conditions with compiling card data?

Reply:
It won't happen as long as these async functions are not updating the same state field

Reply:
What I mean is, if the functions are running async, will there be a possibility that this `compileCardData` runs before all the states have been fully populated? I'm not really privy to all the magics for frontends.

Reply:
ah, the component will always try to match with the latest change in the state which `compileCardData ` used to compile card data